proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -161,9 +285,228 @@ Capabilities required to use above mentioned APIs are as follows:
+
+##### Scenario Description
+1. If the proxy requests `WiFi` status info, SDL Core responds with `SUCCESS` and provides WiFi credentials to app.
+2. User tries to launch the app on HMI.
+3. HMI sends `OnSystemRequest` notification with `CONNECT_WIFI` request type.
+4. HMI shows `Waiting for WiFi Connection. Please check your device when it is safe to do so. Some devices may need your permissions to connect Car WiFi.` pop-up with a timeout of 5 seconds.
+
+##### Post Conditions 
+
+1. WiFi credentials shared with app.
+2. The proxy connects to WiFi using credentials provided by HMI.
+
+##### Exceptions
+
+##### **Exception 1: HMI pop-up `Waiting for WiFi Connection` timeouts**
+
+This case would arise if the WiFi connection APIs fail for any reason.
+
+1. User tries to launch the app on HMI.
+2. HMI shows `Connect WiFi to use this App` pop-up with a timeout of 5 seconds.
+
+#### **D. Driver distraction is ON**
+
+This case is applicable when driver distraction is ON.
+
+##### Scenario Description
+
+1. User tries to launch the app on HMI.
+2. HMI shows `Connect WiFi to use this App` pop-up with a timeout of 5 seconds.
+
+##### Post Conditions 
+
+1. User manually connects WiFi to use app.
+
+#### **E. WiFi auto-connect feature not enabled for app**
+
+This case is applicable when the mobile device does not support WiFi connection APIs or the app developer configures it OFF.
+
+##### Scenario Description
+
+1. User tries to launch the app on HMI.
+2. HMI shows `Connect WiFi to use this App` pop-up with a timeout of 5 seconds.
+
+##### Post Conditions 
+
+1. User manually connects WiFi to use app.
+
+#### Additional Queries
+
+1. Should mobile receive Wifi credentials if it is already connected by Wifi transport?
+- Yes, the proxy will receive WiFi status notifications even when it is connected to WiFi. This would help in cases when WiFi is disabled and enabled again due to some reasons. The proxy can retry connecting using credentials available.
+
+2. If new application request WiFi updates, should it immediately receive cached WiFi credentials?
+- Yes, if allowed by policy, the new app should receive cached WiFi credentials.
+
+3. Should cached Wifi credentials be stored on SDL storage across ignition cycles?
+- No. HMI should always update SDL core about WiFi status changes using `OnWiFiTransportStatusUpdate` HMI API. 
 
 ## Alternatives considered
 
-Leaving the current behavior of SDL over WiFi transport "as is". However, implementing this proposal will make use of Apps easier with WiFi transport.
+1. The proxy can know the status of the current WiFi connection using APIs provided in Android and iOS SDK. However, these APIs require additional permissions and entitlements. 
+
+2. Instead of depending on `Transport Event Update` to know WiFi connection state, `subscription` methods could be provided in mobile API, as defined below. However, this would increase the number of available RPCs for WiFi.
+
+```

```

Comment:
Same comment as above regarding code block formatting here.

Reply:
Added changes in [this ](https://github.com/smartdevicelink/sdl_evolution/pull/819/commits/8cecbd20a37e7db3d90e66bb3a338fabd1aa3c93)commit.